internal/web/context.go

```diff
@@ -0,0 +1,22 @@
+package web
+
+import (
+	"github.com/labstack/echo/v4"
+	"go.uber.org/zap"
+)
+
+type AppContext struct {
```

Comment:
The AppContext struct embeds echo.Context, which can potentially lead to confusion with method chains. It's better to compose the AppContext with a field for echo.Context instead. This helps avoid method name collisions and clarifies the context usage.